test/saithrift/tests/sail3.py

```diff
@@ -1324,4 +1325,159 @@ def runTest(self):
             attr = sai_thrift_attribute_t(id=SAI_PORT_ATTR_PORT_VLAN_ID, value=attr_value)
             self.client.sai_thrift_set_port_attribute(port1, attr)
             self.client.sai_thrift_set_port_attribute(port2, attr)
-
+@group('lag')            
+@group('l3')
+class L3MultipleLagTest(sai_base_test.ThriftInterfaceDataPlane):
+v4_enabled = 1
+    v6_enabled = 1
+    ip_mask = '255.255.255.0'
+    addr_family = SAI_IP_ADDR_FAMILY_IPV4
+    lag_members = []
+    lags = []
+    lags_rifs = []
+    neighbors = []
+    routes = []
+    vr_id = 0
+    mac_action = SAI_PACKET_ACTION_FORWARD
+    src_port = 0
+    mac_pool = ['00:11:22:33:44:50',
+               '00:11:23:33:44:51',
+               '00:11:24:33:44:52',
+               '00:11:25:33:44:53',
+               '00:11:26:33:44:54',
+               '00:11:27:33:44:55',
+               '00:11:28:33:44:56',
+               '00:11:29:33:44:57',
+               '00:11:30:33:44:58',
+               '00:11:31:33:44:59',
+               '00:11:32:33:44:60',
+               '00:11:33:33:44:61',
+               '00:11:34:33:44:62',
+               '00:11:35:33:44:63',
+               '00:11:36:33:44:64',
+               '00:11:37:33:44:65',
+               '00:11:38:33:44:66']
+    
+    
+    def config_for_lags(self, num_of_lags, port_list):
+        print port_list
+        mac =''
+        for i in xrange(num_of_lags):
+            self.lags.append(self.client.sai_thrift_create_lag([]))
+        for i in xrange(16):
+            self.lag_members.append(sai_thrift_create_lag_member(self.client, self.lags[i % num_of_lags], port_list[i]))
+        for i in xrange(num_of_lags):
+            self.lags_rifs.append(sai_thrift_create_router_interface(self.client, self.vr_id, 1, self.lags[i], 0, self.v4_enabled, self.v6_enabled, mac))
+        for i in xrange(16):
+            sai_thrift_create_neighbor(self.client, self.addr_family, self.lags_rifs[i%len(self.lags_rifs)], "10.10.%s.1" % str(i+1), self.mac_pool[i])
+            sai_thrift_create_route(self.client, self.vr_id, self.addr_family, "10.10.%s.0" % str(i+1), self.ip_mask, self.lags_rifs[i% len(self.lags_rifs)])
```

Comment:
there is one neighbor for one rif, and one directly reachable route for one neighbor. the range should be number_of_lags, not 16.